hydra-main/src/main/java/com/addthis/hydra/minion/Minion.java

```diff
@@ -581,10 +581,10 @@ private boolean updateTaskMeta(String jobID, File taskRoot) throws IOException {
         }
     }
 
-    void writeState() {
+    void writeState(boolean append) {
```

Comment:
Is append the correct thing to do? Minion state is a JSON object. If you keep appending, wont that result in invalid json in the file? Have you tested that?